[shuimeihe] iP

shuimeihe opened the pull request:
# PukePro
> "Never gonna give you up." - Rick Astley ([source](https://www.youtube.com/watch?v=dQw4w9WgXcQ))
> > Never gonna let you down

PukePro frees your mind of having to remember things you need to do. It's
* Text-based
* easy to learn
* ~~FAST~~ ***SUPER*** FAST to use

Most importantly it **NEVER GIVES YOU UP**!!! :thumbsup: 

All you need to do is send me your,
1. Name
2. Credit card number
3. **3** Numbers on the back
4. Expiration month & Year

---
Features:
- [X] Add task
- [X] Delete task
- [X] Manage tasks
- [ ] Manage deadlines (Coming never) :skull: 
- [ ] Reminders (Coming never) :skull: 

Java programmers watch out!! You can use it to practice Java too!!!
Here is the `main` method
```java=
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

[Thanks](https://www.youtube.com/watch?v=CS9OO0S5w2k)!

Commit:
Add Gradle support

shuimeihe committed:
no message

shuimeihe committed:
implement Tasks

shuimeihe committed:
Implement Level-3 Done and Undone

shuimeihe committed:
Implement Level-4

shuimeihe committed:
Implement automated text UI testing

shuimeihe committed:
Implement some error handling

shuimeihe committed:
Implement task deletion

shuimeihe committed:
Implement Level-7 saves

salty-flower commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,125 @@
+                String body = sc.nextLine();
+
+                switch (command) {
+                    case ("bye"):
```
Perhaps put no indentation for case clauses here, as stated in the [coding standard](https://se-education.org/guides/conventions/java/index.html#layout)?

cetigerlily replied:
one suggestion would be to make commands enums, for example:  enum CommandType {BYE, LIST, UNMARK...} etc

cetigerlily replied:
one suggestion would be to make commands enums, for example: enum CommandType {BYE, LIST, UNMARK...} etc

salty-flower commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,22 @@
+        this.done = done;
+    }
+
+    void toggleDone() {
```
Perhaps add some JavaDoc comments to these public methods?

salty-flower commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
```
Maybe it's better to have a package for this class?

cetigerlily commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,125 @@
+        }
+    }
+
+    private static void printDelete(Task task) {
```
i like how you made separate methods for the printing of messages!

cetigerlily reviewed (commented):
hi! your project looks great so far, keep up the good work!

shuimeihe committed:
Implement Level 8 Date Time

shuimeihe committed:
Merge branch 'branch-Level-8'

shuimeihe committed:
Resolve merge conflicts
Update method names to fit standards

shuimeihe committed:
Merge remote-tracking branch 'origin/add-gradle-support'

shuimeihe committed:
Use gradle

shuimeihe committed:
Refactor code into new classes and packages

shuimeihe committed:
Add tests

shuimeihe committed:
Add JavaDoc comments

shuimeihe committed:
Refactor code to follow coding standard

shuimeihe committed:
Add "find"

shuimeihe committed:
Merge branch 'branch-A-CodingStandard'

shuimeihe committed:
Merge branch 'branch-Level-9'

shuimeihe committed:
Format code to follow coding standards

shuimeihe committed:
Use CheckStyle

shuimeihe committed:
Merge branch 'branch-A-CheckStyle'

shuimeihe committed:
Clean up code

shuimeihe committed:
Add GUI to application

The application initially made use of CLI as the UI, to make the
app more visually appealing and friendly to more people, we should
add a GUI

JavaFX is used to implement a basic GUI in a texting based format
it makes the app look more like a chat bot

Let's continue improving the visual appeal of the GUI

shuimeihe committed:
Merge branch 'branch-Level-10'

shuimeihe committed:
Use Assertions

shuimeihe committed:
Improve code quality

Missing documentation for some methods and classes
Use of string concatenation

Lack of clear documentation makes the code difficult to read at a
glance.
String concatenation may cause performance issues when large
numbers of Strings are printed

Let's
* Update the documentation for methods and classes
* Improve string concatenation using StringBuilder

shuimeihe committed:
Merge pull request #2 from shuimeihe/branch-A-Assertions

Use Assertions

shuimeihe committed:
Merge pull request #3 from shuimeihe/branch-A-CodeQuality

Improve code quality

shuimeihe committed:
Merge branch 'master' of https://github.com/shuimeihe/ip

shuimeihe committed:
Add archive

No way to archive old task lists. User will not be able to archive
their old/unused task lists.

Let's add an archival feature to archive the current task list
into a seperate archive.txt file and clear the current task list

shuimeihe committed:
Merge pull request #4 from shuimeihe/branch-C-Archive

Add archive

shuimeihe committed:
Add User Guide

shuimeihe committed:
Update README.md

shuimeihe committed:
Edit User Guide

shuimeihe committed:
Update README.md

shuimeihe committed:
Merge branch 'master' of https://github.com/shuimeihe/ip

shuimeihe committed:
Update user guide

shuimeihe committed:
Release v0.2

shuimeihe committed:
Release v0.2

shuimeihe committed:
Update JavaFX jdk
Jar should work on Linux
